### PR TITLE
Feature/occupation input component

### PR DESCRIPTION
**Pull request by birom**
Auto complete input compment for occupations.
input-screenshot-01

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,32 +7,32 @@
...
+    "@ng-bootstrap/ng-bootstrap": "^1.0.0-alpha.28",
+    "core-js": "^2.4.1",
     "jquery": "3.2.1",
     "ng-jhipster": "0.2.3",
     "ng2-webstorage": "1.8.0",
     "ngx-cookie": "1.0.0",
     "ngx-infinite-scroll": "0.4.1",
     "reflect-metadata": "0.1.10",
-    "rxjs": "5.4.0",
+    "rxjs": "^5.1.0",
```
Comment (michael-wirth) [inline package.json]: is version downgrading correct?

File: package.json [hunk 2]
```diff
@@ -86,10 +86,10 @@
     "tslint": "4.5.1",
     "tslint-eslint-rules": "^4.1.1",
     "tslint-loader": "3.5.3",
-    "typescript": "2.3.3",
+    "typescript": "~2.3.3",
     "web-app-manifest-loader": "0.1.1",
-    "webpack": "2.6.1",
-    "webpack-dev-server": "2.4.5",
+    "webpack": "~2.4.0",
```
Comment (michael-wirth) [inline package.json]: downgrading?

File: src/main/webapp/app/job-search/job-search-filter/job-search-filter.component.ts [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 })
 export class JobSearchFilterComponent implements OnInit {
 
-  constructor() { }
+    constructor() {
+    }
 
-  ngOnInit() {
-  }
+    ngOnInit() {
```
Comment (michael-wirth) [inline src/main/webapp/app/job-search/job-search-filter/job-search-filter.component.ts]: remove this method if the `OnInit` is not used

File: src/main/webapp/app/job-search/job-search-filter/job-search-filter.component.ts [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
-  templateUrl: './job-search-filter.component.html',
-  styles: []
+    selector: 'jr2-job-search-filter',
+    templateUrl: './job-search-filter.component.html',
+    styles: []
 })
 export class JobSearchFilterComponent implements OnInit {
 
-  constructor() { }
+    constructor() {
```
Comment (michael-wirth) [inline src/main/webapp/app/job-search/job-search-filter/job-search-filter.component.ts]: empty constructor can also be removed

File: src/test/javascript/spec/app/job-search/shared/occupation.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            imports: [JobroomTestModule],
+            providers: [
+                OccupationService,
+                { provide: TranslateService, useValue: { currentLang: 'de' } }
+            ]
+        });
+    });
+
+    describe('fetchSuggestions', () => {
+        // GIVEN
```
Comment (michael-wirth) [inline src/test/javascript/spec/app/job-search/shared/occupation.service.spec.ts]: GIVEN seems to be at the wrong place. it should be places inside the `it` test. 

Here i'd rather user `PREPARE` or `SETUP`

File: package.json (hunk 1 above)
Comment (birom) [inline package.json]: It is not downgrading. Caret updates to the most recent minor version.

File: package.json (hunk 2 above)
Comment (birom) [inline package.json]: Tilde updates to the most recent patch version.

File: package.json (hunk 1 above)
Comment (michael-wirth) [inline package.json]: ok, we need to document this in the wiki.
otherwise this cause problems in the future

Comment (birom) [inline package.json]: But you are right it is confusing to upgrade with a smaller number... I am going to fix it.

**Pull request merged**